protocol/src/model/packets/play.rs

```diff
@@ -34,7 +34,7 @@ pub struct LoginPlay {
     pub game_mode: GameMode,
     pub prev_game_mode: PreviousGameMode,
     pub dimension_names: Array<Identifier>,
-    pub registry_codec: NbtSerde<HashMap<String, Nbt>>,
+    pub registry_codec: Nbt,
```

Comment:
here this was modified from a HashMap to raw NBT, removing the guarantee of it being a compound